doc/cpp/domdec.rst

```diff
@@ -7,136 +7,127 @@ Domain decomposition
-        thread safe.
-
-    .. cpp:member:: int num_domains
-
-        Number of domains that the model is distributed over.
-
-    .. cpp:member:: int domain_id
-
-        The index of the local domain.
+        directly constructed by a user, the following conditions must be met,
+        if not, an exception will be thrown:
+
+        * Every cell in the model appears once in one and only one cell
+          :cpp:class:`group <group_description>` on one and only one local
+          :cpp:class:`domain_decomposition` object.
+        * The total number of cells across all cell
+          :cpp:class:`groups <group_description>` on all
+          :cpp:class:`domain_decomposition` objects must match the total
+          number of cells in the :cpp:class:`recipe`.
+        * Cells that are connected via gap-junction must be present in the
+          same cell :cpp:class:`group <group_description>`.
+
+    .. cpp:function:: domain_decomposition(const recipe& rec, const context& ctx, const std::vector<group_description>& groups)
+
+        The constructor takes:
+
+        *   a :cpp:class:`arb::recipe` that describes the model;
+        *   a :cpp:class:`arb::context` that describes the hardware resources;
+        *   a vector of :cpp:class:`arb::group_description` that contains the indices of the cells
+            to be executed on the local rank, categorized into groups.
+
+        It's expected that a different :cpp:class:`arb::domain_decomposition` object will be constructed on
+        each rank in a distributed simulation containing that selected cell groups for that rank.
+        For example, in a simulation of 10 cells on 2 MPI ranks where cells {0, 2, 4, 6, 8} of kind
+        :class:`cable_cell` are meant to be in a single group executed on the GPU on rank 0;
+        and cells {1, 3, 5, 7, 9} of kind :class:`lif_cell` are expected to be in a single group executed
+        on the CPU on rank 1:
+
+        Rank 0 should run:
+
+        .. code-block:: c++
+
+            std::vector<arb::group_description> groups = {
+                {arb::cell_kind::cable, {0, 2, 4, 6, 8}, arb::backend_kind::gpu}
+            };
+            auto decomp = arb::domain_decomposition(recipe, context, groups);
+
+        And Rank 1 should run:
+
+        .. code-block:: c++
+
+            std::vector<arb::group_description> groups = {
+                {arb::cell_kind::lif,   {1, 3, 5, 7, 9}, arb::backend_kind::multicore}
+            };
+            auto decomp = arb::domain_decomposition(recipe, context, groups);
+
+        .. _domdecnotes:
+        .. Important::
+            Constructing a balanced :cpp:class:`domain_decomposition` quickly
+            becomes a difficult task for large and diverse networks. This is why
+            arbor provides :ref:`load balancing algorithms<domdecloadbalance>`
+            that automatically generate a :cpp:class:`domain_decomposition` from
```

Comment:
Typo: should be generate*s*

Reply:
it's load balancing algorithms in plural. I know we only have one at the moment, but the plan has always been to add more.

Reply:
Ok, that was me reading `partition_load_balance`. The curse of a prior.